src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -217,60 +216,70 @@ private ScriptEngineService getScriptEngineServiceForFileExt(String fileExtensio
     /**
      * Checks if a script can be executed and compiles it if needed, or returns the previously compiled and cached script.
      */
-    public CompiledScript compile(String lang,  String script, ScriptType scriptType, ScriptContext scriptContext) {
-        assert script != null;
-        assert scriptType != null;
-        assert scriptContext != null;
+    public Script compile(Script script) {
+        assert script.key != null;
```

Comment:
Done.  The reason I didn't do this before was because the places I was using the Script object didn't necessarily have all of those pieces.